Packages/io.chainsafe.web3-unity/Samples~/Web3.Unity/Scripts/RampIntegration/RampIntegrationSample.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using ChainSafe.GamingSdk.RampIntegration;
+using ChainSafe.GamingSdk.ScriptableObjects;
+using UnityEngine;
+
+public class RampIntegrationSample : MonoBehaviour
+{
+    [SerializeField] private RampData rampData;
+    private RampChainsafeIntegrationBase _ramp;
+
+    private void Awake()
+    {
+#if UNITY_IOS
+        _ramp = new RampChainsafeIntegrationiOS(rampData);
+#endif
```

Comment:
So while this looks simple & clean, I would move the Ramp object reference to the Web3 infrastructure. This way it would utilize dependency injection (we could use the current player address for example) and it would be accessible just like any other component provided by the SDK. Kinda like this:
`web3.RampExchanger().Buy(...)` or `web3.RampExchanger().Open(...)` or `web3.RampExchanger().Show(...)` idk
_See src/ChainSafe.Gaming.Gelato/GelatoExtensions.cs_

Choosing the required implementation can also be done silently so that users won't have to care about it.

Reply:
You would have to provide RampData to OpenRamp method, because there will only be one instance of RampExchanger.

Reply:
My biggest concern in doing that is -> Can we call native iOS code with extern trough our gaming sdk?

Reply:
Do we have to?

Reply:
Unfortunately yes. The essential part here is the "MonoPInvokeCallback" in the RampExchangeriOS class that's actually a callback that is being called by the native iOS code.

Reply:
I did update the project structure tho. I'm not doing the creation of the RampExchanger subclasses in the Sample class, but I 've made the factory that will do that for each individual platform.

Reply:
Sorry, I don't understand. Aren't you already doing it for iOS?

Reply:
Yes, so you should update your webGL branch 😁